scvi/inference/posterior.py

```diff
@@ -1826,16 +1766,17 @@ def knn_purity(self) -> torch.Tensor:
 
     @torch.no_grad()
     def clustering_scores(self, prediction_algorithm: str = "knn") -> Tuple:
-        if self.gene_dataset.n_labels > 1:
+        if self.gene_dataset.adata.uns["scvi_summary_stats"]["n_labels"] > 1:
             latent, _, labels = self.get_latent()
             if prediction_algorithm == "knn":
                 labels_pred = KMeans(
-                    self.gene_dataset.n_labels, n_init=200
+                    self.gene_dataset.adata.uns["scvi_summary_stats"]["n_labels"],
+                    n_init=200,
                 ).fit_predict(
                     latent
                 )  # n_jobs>1 ?
```

Comment:
this commented portion should be deleted if we don't use it actively (I used this code once or twice)